include/swift/AST/Decl.h

```diff
@@ -8452,7 +8453,8 @@ class MacroExpansionDecl : public Decl {
   SourceLoc LeftAngleLoc, RightAngleLoc;
   ArrayRef<TypeRepr *> GenericArgs;
   ArgumentList *ArgList;
-  ArrayRef<Decl *> Rewritten;
+  // An implicit brace statement serving as the container for the expansion.
+  BraceStmt *Rewritten = nullptr;
```

Comment:
Brace statements imply a nested scope, which probably isn't what we want semantically. Can this become an `ArrayRef<ASTNode>` to mimic the `[CodeBlockItemSyntax]` we recently agreed upon for the macro implementation interface?

Reply:
Makes sense.  I was relying on `BraceStmt` merely as a convenient container, and the `BraceStmt` was never visited as a child of `MacroExpransionDecl`; ASTWalker directly walks the `BraceStmt`'s children instead. I think it's better than `ArrayRef<ASTNode>` because we can distinguish between `nullptr` (expansion failure) and an empty list from the return of `swift::expandFreestandingMacro()`. Otherwise I would need to make it return a `bool` and produce indirect results into a `ASTNode` vector. Thoughts?